lib/plugins/aws/package/lib/mergeCustomProviderResources.js

```diff
@@ -12,11 +12,72 @@ module.exports = {
       this.serverless.service.resources.Outputs = {};
     }
 
+    const extensions = this.serverless.service.resources
+      ? this.serverless.service.resources.extensions
+      : null;
+    if (extensions) {
+      delete this.serverless.service.resources.extensions;
+    }
+
     _.merge(
       this.serverless.service.provider.compiledCloudFormationTemplate,
       this.serverless.service.resources
     );
 
+    if (extensions) {
+      const template = this.serverless.service.provider.compiledCloudFormationTemplate;
+
+      _.forOwn(extensions, (resourceDefinition, resourceName) => {
```

Comment:
Let's resort to lodash, only if we do not have native counterpart. In that case it'll be nicer to construct it with `for..of` as:

```javascript
for (const [resourceName, resourceDefinition] of _.entries(extensions)) {
...
```

We'd be able to replace `_.entries` with `Object.entries` once we drop support for Node.js v6 (and that should happen shortly)